### PR TITLE
[VL] Rewrite collect_set and collect_list aggregate function

### PR DESCRIPTION
**Pull request by ulysses-you**
## What changes were proposed in this pull request?

- Add `IsNotNull(partial_in)` to skip null value before going to native collect_set
- Add `If(IsNull(result), CreateArray(Seq.empty), result)` to replace null to empty array

## How was this patch tested?

add test

pass `SPARK-17641: collect functions should not collect null values`

### CODE REVIEW

File: backends-velox/src/test/scala/io/glutenproject/execution/VeloxAggregateFunctionsSuite.scala [hunk 1]
```diff
@@ -699,6 +699,46 @@ abstract class VeloxAggregateFunctionsSuite extends VeloxWholeStageTransformerSu
...
 
+  test("test collect_set") {
+    runQueryAndCompare("SELECT array_sort(collect_set(l_partkey)) FROM lineitem") {
+      df =>
+        {
+          assert(
+            getExecutedPlan(df).count(
+              plan => {
+                plan.isInstanceOf[HashAggregateExecTransformer]
+              }) == 2)
```
[the remark this answers is not in the document]
Comment (zhztheplayer) [reply backends-velox/src/test/scala/io/glutenproject/execution/VeloxAggregateFunctionsSuite.scala:710]: Oops. Somehow I omitted this code. Thanks for quoting.

**Comment by ulysses-you** [comment]
cc @zhztheplayer @PHILO-HE @rui-mo  @liujiayi771 thank you

File: gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala [hunk 2]
```diff
@@ -54,9 +54,7 @@ class VeloxTestSettings extends BackendTestSettings {
       "SPARK-32038: NormalizeFloatingNumbers should work on distinct aggregate",
       // Replaced with another test.
       "SPARK-19471: AggregationIterator does not initialize the generated result projection" +
-        " before using it",
```
Comment (liujiayi771) [inline gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:57]: Could you also test whether the test case for "SPARK-31993: concat_ws in agg function with plenty of string/array types columns" can now be included?

**Comment by liujiayi771** [comment]
https://github.com/oap-project/gluten/issues/4763

File: gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala (hunk 2 above)
Comment (ulysses-you) [reply gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:57]: thank you for the reminder, enable that test

**Review by PHILO-HE — APPROVED** [review]
Thank for your fix! It's nice to introduce such rule before inconsistency behavior is fixed in velox.

**Pull request merged**